packages/main/src/requests/request.ts

```diff
@@ -43,9 +43,11 @@ export class RequestUrl {
     public task: Task,
     public apiKey: string,
     public stream: boolean,
+    public requestOptions: RequestOptions,
   ) {}
   toString(): string {
-    let url = `${BASE_URL}/${API_VERSION}/${this.model}:${this.task}`;
+    const apiVersion = this.requestOptions?.apiVersion || DEFAULT_API_VERSION;
```

Comment:
Should we validate `this.requestOptions?.apiVersion` to check if it is either `v1` or `v1beta`?

Reply:
No, I want to future-proof it so any version is allowed, so we don't have to update the SDK every time a new version becomes available on the backend.